edx_notifications/stores/sql/models.py

```diff
@@ -402,3 +428,13 @@ def load_from_data_object(self, notification_timer):
         self.executed_at = notification_timer.executed_at
         self.err_msg = notification_timer.err_msg
         self.results = DictField.to_json(notification_timer.results)
+
+
+@receiver(pre_delete, sender=SQLUserNotification)
+def archive_deleted_user_notification(sender, instance, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Archiving the deleted user notifications.
+    """
+    notification_archive_obj = SQLUserNotificationArchive()
```

Comment:
Can you add a configuration flag here to check if the installation wishes to actually archive notifications. I say this because many installations of Open edX might not wishes to keep all these in a database, since the database table will grow very large over time.

Can you put this configuration in const.py and have it default to "False" (don't archive messages by default)? Thanks.